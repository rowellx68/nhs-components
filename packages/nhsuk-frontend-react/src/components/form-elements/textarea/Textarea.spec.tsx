import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { composeStory } from '@storybook/react';
import meta, {
  Default as DefaultStory,
  WithRows as WithRowsStory,
  WithError as WithErrorStory,
} from './Textarea.stories';

const Default = composeStory(DefaultStory, meta);
const WithRows = composeStory(WithRowsStory, meta);
const WithError = composeStory(WithErrorStory, meta);

it('should render a textarea element', () => {
  const { container } = render(<Default />);

  expect(container.querySelector('textarea')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render a textarea element with a hint', () => {
  const { container } = render(<Default hint="Hint" />);

  expect(container.querySelector('textarea')).toBeInTheDocument();
  expect(container.querySelector('.nhsuk-hint')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render a textarea element with an error', () => {
  const { container } = render(<WithError />);

  expect(container.querySelector('textarea')).toBeInTheDocument();
  expect(container.querySelector('.nhsuk-error-message')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it('should render a textarea element with rows', () => {
  const { container } = render(<WithRows />);

  expect(container.querySelector('textarea')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
