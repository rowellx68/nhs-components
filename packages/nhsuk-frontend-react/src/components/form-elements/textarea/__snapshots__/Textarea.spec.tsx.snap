// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render a textarea element 1`] = `
<div>
  <div
    class="nhsuk-character-count"
    data-module="nhsuk-character-count"
  >
    <div
      class="nhsuk-form-group"
    >
      <label
        class="nhsuk-label"
        for="textarea_:r0:"
        id="textarea_:r0:--label"
      >
        Can you provide more detail?
      </label>
      <div
        class="nhsuk-hint"
        id="textarea_:r0:_:r1:--hint"
      >
        Do not include personal or financial information, for example, your National Insurance number or credit card details.
      </div>
      <textarea
        aria-describedby="textarea_:r0:_:r1:--hint"
        aria-labelledby="textarea_:r0:--label"
        as="textarea"
        class="nhsuk-textarea nhsuk-js-character-count"
        id="textarea_:r0:"
      />
      <div
        class="nhsuk-hint nhsuk-character-count__message"
        id="textarea_:r0:-info"
      />
    </div>
  </div>
</div>
`;

exports[`should render a textarea element with a hint 1`] = `
<div>
  <div
    class="nhsuk-character-count"
    data-module="nhsuk-character-count"
  >
    <div
      class="nhsuk-form-group"
    >
      <label
        class="nhsuk-label"
        for="textarea_:r2:"
        id="textarea_:r2:--label"
      >
        Can you provide more detail?
      </label>
      <div
        class="nhsuk-hint"
        id="textarea_:r2:_:r3:--hint"
      >
        Hint
      </div>
      <textarea
        aria-describedby="textarea_:r2:_:r3:--hint"
        aria-labelledby="textarea_:r2:--label"
        as="textarea"
        class="nhsuk-textarea nhsuk-js-character-count"
        id="textarea_:r2:"
      />
      <div
        class="nhsuk-hint nhsuk-character-count__message"
        id="textarea_:r2:-info"
      />
    </div>
  </div>
</div>
`;

exports[`should render a textarea element with an error 1`] = `
<div>
  <div
    class="nhsuk-character-count"
    data-module="nhsuk-character-count"
  >
    <div
      class="nhsuk-form-group nhsuk-form-group--error"
    >
      <label
        class="nhsuk-label"
        for="textarea_:r4:"
        id="textarea_:r4:--label"
      >
        Why can't you provide a National Insurance number?
      </label>
      <span
        class="nhsuk-error-message"
        id="textarea_:r4:--error-message"
      >
        <span
          class="nhsuk-u-visually-hidden"
        >
          Error: 
        </span>
        You must provide an explanation.
      </span>
      <textarea
        aria-describedby="textarea_:r4:--error-message"
        aria-labelledby="textarea_:r4:--label"
        as="textarea"
        class="nhsuk-textarea nhsuk-textarea--error nhsuk-js-character-count"
        id="textarea_:r4:"
        rows="5"
      />
      <div
        class="nhsuk-hint nhsuk-character-count__message"
        id="textarea_:r4:-info"
      />
    </div>
  </div>
</div>
`;

exports[`should render a textarea element with rows 1`] = `
<div>
  <div
    class="nhsuk-character-count"
    data-module="nhsuk-character-count"
  >
    <div
      class="nhsuk-form-group"
    >
      <label
        class="nhsuk-label"
        for="textarea_:r6:"
        id="textarea_:r6:--label"
      >
        Can you provide more detail?
      </label>
      <div
        class="nhsuk-hint"
        id="textarea_:r6:_:r7:--hint"
      >
        Do not include personal or financial information, for example, your National Insurance number or credit card details.
      </div>
      <textarea
        aria-describedby="textarea_:r6:_:r7:--hint"
        aria-labelledby="textarea_:r6:--label"
        as="textarea"
        class="nhsuk-textarea nhsuk-js-character-count"
        id="textarea_:r6:"
        rows="10"
      />
      <div
        class="nhsuk-hint nhsuk-character-count__message"
        id="textarea_:r6:-info"
      />
    </div>
  </div>
</div>
`;
